File: JS/Dynamic Programming/198. House Robber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function(nums) {
    const dp = [0, nums[0]];
    const n = nums.length;
    for(let i = 2; i <= n; i++) {
        dp[i] = Math.max(nums[i - 1] + dp[i - 2], dp[i - 1]);
    }
    return dp[n];
};

File: Golang/DynamicProgramming/198.House-robber.go
package main

func rob(nums []int) int {
	dp := make([]int, len(nums)+1)
	dp[0] = 0
	dp[1] = nums[0]
	for i := 2; i <= len(nums); i++ {
		with := dp[i-2] + nums[i-1]
		without := dp[i-1]
		if with > without {
			dp[i] = with
		} else {
			dp[i] = without
		}
	}
	return dp[len(nums)]
}
